alma/alma_test.go

```diff
@@ -0,0 +1,87 @@
+package alma_test
+
+import (
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/aquasecurity/vuln-list-update/alma"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/xerrors"
+)
+
+func TestUpdate(t *testing.T) {
+	tests := []struct {
+		name          string
+		version       string
+		inputJSONFile string
+		expectedError error
+	}{
+		{
+			name:          "happy path",
+			version:       "8",
+			inputJSONFile: "testdata/errata.json",
+			expectedError: nil,
+		},
+		{
+			name:          "sad path, invalid release version",
+			version:       "9",
+			inputJSONFile: "",
+			expectedError: xerrors.Errorf("failed to update security advisories of AlmaLinux 9: %w", errors.New("failed to fetch security advisories from AlmaLinux")),
+		},
+		{
+			name:          "sad path, invalid json",
+			version:       "8",
+			inputJSONFile: "testdata/invalid.json",
+			expectedError: xerrors.Errorf("failed to update security advisories of AlmaLinux 8: %w", errors.New("failed to unmarshal json")),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tsErrataURL := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
nit: this is an instance of test server and not URL. `ts` is enough or `errataTestSever` might be more descriptive if you prefer.

```suggestion
			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Reply:
The name of the variable was misleading.
Thanks for pointing it out.
I've changed it to ts.